src/sheets/updateItem.js

```diff
@@ -0,0 +1,45 @@
+import logger from '../logger'
+import {getValues, batchUpdateValues} from './sheets.js'
+import {sheets} from './constants'
+
+export async function updateItem(
+  isCompanyItem,
+  items,
+  getCompanyItemData,
+  getPersonalItemData,
+) {
+  const sheet = isCompanyItem ? sheets.companyOrders : sheets.personalOrders
+
+  const itemsMap = new Map(items.map((item) => [item.id, item]))
+
+  const itemIdsFromSheet = await getValues(sheet.idRange)
+
+  const data = itemIdsFromSheet.reduce((result, itemId, index) => {
+    if (itemsMap.has(itemId[0])) {
+      const rowIndex = index + sheet.rowOffset + 1
+
+      let itemData
+      if (isCompanyItem) {
+        itemData = getCompanyItemData
+          ? getCompanyItemData(sheet.name, rowIndex, itemsMap.get(itemId[0]))
```

Comment:
1. After checking the code, `getCompanyItemData` should always be defined.
2. IMO rather than checking if `getCompanyItemData` (or `getPersonalItemData`) is there, it would be better to have a default value for it (a simple function like `() => null`) https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters
This will make the code more compact, then we can
```
const itemData = isCompanyItem
  ? getCompanyItemData(...)
  : getPersonalItemData(...)
```